### PR TITLE
Starts to fix Issue #651: begins with deprecation warning

### PR DESCRIPTION
**Pull request by cryptomail**
Starts to fix Issue #651: begins with deprecation warning when using password

### CODE REVIEW

File: tests/test_api/test_other_zendesk_objects.py [hunk 1]
```diff
@@ -428,3 +429,17 @@ def test_users_me(self):
...
             self.assertNotEqual(me, None, "me is valid")
             self.assertNotEqual(me.email, "", "email is valid in me")
+
+class TestPasswordDeprecation(ZenpyApiTestCase):
+    __test__ = True
+    def test_password_failure(self):
+        with self.assertRaises(ZenpyException):
+            zenpy_client = Zenpy(subdomain="party", email="face@toe", password="Yer", password_treatment_level="error")
+            self.assertTrue(False)
+    def test_password_passes(self):
```
Comment (token-cjg) [inline tests/test_api/test_other_zendesk_objects.py:439]: nit: missing whitespace between tests

**Review by token-cjg — APPROVED** [review]

File: tests/test_api/test_other_zendesk_objects.py [hunk 2]
```diff
@@ -428,3 +429,17 @@ def test_users_me(self):
             me = self.zenpy_client.users.me()
             self.assertNotEqual(me, None, "me is valid")
             self.assertNotEqual(me.email, "", "email is valid in me")
+
+class TestPasswordDeprecation(ZenpyApiTestCase):
+    __test__ = True
+    def test_password_failure(self):
+        with self.assertRaises(ZenpyException):
+            zenpy_client = Zenpy(subdomain="party", email="face@toe", password="Yer", password_treatment_level="error")
+            self.assertTrue(False)
```
Comment (sircpl) [inline tests/test_api/test_other_zendesk_objects.py:438]: you can probably get rid of this, asserting on the raise is good enough

File: tests/test_api/test_other_zendesk_objects.py [hunk 3]
```diff
@@ -428,3 +429,17 @@ def test_users_me(self):
...
+    def test_password_passes(self):
+        zenpy_client = Zenpy(subdomain="party", email="face@toe", password="Yer", password_treatment_level="warning")
+        self.assertTrue(True)
```
Comment (sircpl) [inline tests/test_api/test_other_zendesk_objects.py:441]: Does this assertion accomplish anything? Is it possible to assert on the logging instead?

File: tests/test_api/test_other_zendesk_objects.py [hunk 4]
```diff
@@ -428,3 +429,17 @@ def test_users_me(self):
...
+
+    def test_password_passes_no_deprecation(self):
+        zenpy_client = Zenpy(subdomain="party", email="face@toe", password="Yer")
+        self.assertTrue(True)
```
Comment (sircpl) [inline tests/test_api/test_other_zendesk_objects.py:445]: same

File: tests/test_api/test_other_zendesk_objects.py (hunk 1 above)
Comment (cryptomail) [reply tests/test_api/test_other_zendesk_objects.py:439]: willfix

File: tests/test_api/test_other_zendesk_objects.py (hunk 3 above)
Comment (cryptomail) [reply tests/test_api/test_other_zendesk_objects.py:441]: I'll have to see if there's a mock or something that ensures the logger emitted the warning....that was the *RIGHT* way to do it.  I'll fix that.

Comment (cryptomail) [reply tests/test_api/test_other_zendesk_objects.py]: found, fixed.

**Pull request merged**